hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -882,9 +917,9 @@ bool MDMParser::_powerOn(void)
     bool retried_after_reset = false;
     int i = MDM_POWER_ON_MAX_ATTEMPTS_BEFORE_RESET; // When modem not responsive on boot, AT/OK tries 25x (for ~30s) before hard reset
 
-    // FIXME: REMOVE THIS BEFORE MERGING!!!!!!
-    _dev.dev = DEV_SARA_R510;
-    MDM_ERROR("Forcing to R510\r\n");
+    auto otp_ncp_id = platform_primary_ncp_identifier();
+    _dev.dev = fromPlatformNcpIdentifer(otp_ncp_id);
+    MDM_INFO("ncp_id otp value: 0x%X _dev.dev SET AS = 0x%X \r\n", otp_ncp_id, _dev.dev);
```

Comment:
My original comment about `MDM_` macros still stands. For any new code let's use `LOG()`. We also have a long standing story to refactor this: see CH50252.